packages/@lwc/engine-core/src/framework/vm.ts

```diff
@@ -112,8 +112,11 @@ export interface VM<N = HostNode, E = HostElement> {
     /** The owner VM or null for root elements. */
     readonly owner: VM<N, E> | null;
     /** Rendering operations associated with the VM */
-    renderMode: RenderMode;
+    readonly renderMode: RenderMode;
     shadowMode: ShadowMode;
+    /** Transitive support for native Shadow DOM. A component in native mode
+     * transitively opts all of its descendants into native. */
+    readonly nearestSRMode: ShadowMode | null;
```

Comment:
this can now be computed on the spot when constructing the vm based on the owner's values.

Reply:
I'd add a comment here explaining that SR stands for Shadow Root. (Or rename the property.) Otherwise it may not be obvious to the reader.

Reply:
```suggestion
    readonly nearestShadowMode: ShadowMode | null;
```